src/rgw/motr/sync/motr_sync_impl.cc

```diff
@@ -185,3 +185,22 @@ int MotrKVLockProvider::remove_lock(const std::string& lock_name,
   }
   return rc;
 }
+
+// Create a global instance of MotrLock that can be used by caller
+// This needs to be called by a single main thread of the caller.
+std::shared_ptr<MotrSync> g_motr_lock;
+std::shared_ptr<MotrSync>& create_motr_Lock_instance(
```

Comment:
we can rename this function as `get_lock_instance()` and eliminate the other method.

Reply:
ok